### PR TITLE
C++: Add test cases with SAMATE Juliet code snippets to the codeql test suite.

### PR DESCRIPTION
**Pull request by geoffw0**
Add (previously internal) test cases with SAMATE Juliet code snippets to the codeql test suite.  Though we do test against the entire SAMATE Juliet test suite, we get faster and more accurate information about regressions from these tests.

### CODE REVIEW

File: cpp/ql/test/query-tests/Security/CWE/notice.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+Some of the the files in these tests contain source code copied or derived from the public domain "Juliet Test Suite for C/C++" (provided by NIST / SAMATE Team at https://samate.nist.gov/SARD/testsuite.php).  Such tests are typically in subdirectories named "SAMATE".
```
Comment (jbj) [inline cpp/ql/test/query-tests/Security/CWE/notice.txt:2]: Can this file be named `README` instead? Or some other file name that GitHub will consider to be a readme and therefore display when listing the directory.

Comment (geoffw0) [reply cpp/ql/test/query-tests/Security/CWE/notice.txt]: Good idea (there's no need for it to be `notice.txt`, and having it more visible on GitHub seems like a plus).

Done.

**Review by jbj — COMMENTED** [review]
LGTM. I'll merge if @rdmarsh2 hasn't commented by tomorrow.

**Comment by jbj** [comment]
Actually, the language tests failed.

**Comment by geoffw0** [comment]
I think the test failures were caused by the recent subpaths PR.

**Review by jbj — APPROVED** [review]

**Pull request merged**